src/core/client.c

```
@@ -329,6 +333,15 @@ mender_client_init(mender_client_config_t *config, mender_client_callbacks_t *ca
         goto END;
     }
 
+#ifdef CONFIG_MENDER_SERVER_DORMANT_CERTIFICATE
+    int rc = mender_add_dormant_cert();
+    if (0 != rc && -EEXIST != rc) {
+        mender_log_error("Failed to add dormant certificate: %d", rc);
```

Should the client fail here or continue? :thinking:

---

Unsure. What do you think?

---

What do I think?

TLDR; fatal error :ok_hand: 

As a user,

Similar to what we have for Linux there would be an "integration checklist" for Mender MCU, where I have instructions to make some experiments before I send my device out in the field. There, there will be some special build options to verify that the dormant certificate can be added but also that can actually communicate. So, when building in such mode the error here should be fatal but when doing regular build it should not.

As an engineer,

The above can be a follow-up epic on itself. But in the context of this task the safest is to treat it as a fatal error. If an specific device (read: an specific MbedTLS configuration) has issues with the cert is better to fail hard and let the user fix or remove the feature from Mender MCU Kconfig before shipping devices into the field.